dts/arm/silabs/siwg917.dtsi

```diff
@@ -49,6 +46,20 @@
 			reg = <0x46130000 0x1000>;
 		};
 
+		flashctrl0: flash-controller@12000000 {
+			compatible = "silabs,siwx917-flash-controller";
+			reg = <0x12000000 0x200>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@8200000 {
```

Comment:
I wonder if this should actually be moved to the board dts, since the address of the flash changes depending on whether the board uses common (internal) flash or dual flash. For now, I see that the flash driver only supports common flash, so it doesn't really matter, though.